mlflow/azureml/__init__.py

```diff
@@ -4,135 +4,297 @@
-        mlflow_dir = mlflow.utils.file_utils._copy_project(src_path=mlflow_home, dst_path="./")
-        deps = "-d {}".format(mlflow_dir)
-        mlflow_dep = "-e /var/azureml-app/{}".format(mlflow_dir)
+    :param registered_model: The Azure Model that the execution script will load for inference.
+    :return: A reference to the temporary file containing the execution script.
+    """
+    # Azure ML requires the container's execution script to be located
+    # in the current working directory during image creation, so we
+    # create the execution script as a temporary file in the current directory
+    execution_script_file = tempfile.NamedTemporaryFile(
+            dir=os.getcwd(), mode="w", prefix="driver", suffix=".py")
+    execution_script_text = SCORE_SRC.format(
+            model_name=azure_model.name, model_version=azure_model.version)
+    execution_script_file.write(execution_script_text)
+    # This file will be read from the current location of the file pointer, which was
+    # advanced during the write. Therefore, we seek to the beginning of the file so that
+    # its full contents can be read later
+    execution_script_file.seek(0)
+    # Temporary files created using the `tempfile.NamedTemporaryFile` constructor are deleted
+    # when their associated object reference goes out of scope. Therefore, We return a reference to
+    # the execution script file to prevent it from being deleted prematurely
+    return execution_script_file
+
+
+def _create_dockerfile(output_path, mlflow_path=None):
+    """
+    Creates a Dockerfile containing additional Docker build steps to execute
+    when building the Azure container image. These build steps perform the following tasks:
 
-    with open("requirements.txt", "w") as f:
-        f.write(mlflow_dep + "\n")
+    - Install MLflow
 
-    shutil.copytree(src=model_path, dst="model")
+    :param output_path: The path where the Dockerfile will be written.
+    :param mlflow_path: Path to a local copy of the MLflow GitHub repository. If specified, the
+                        Dockerfile command for MLflow installation will install MLflow from this
+                        directory. Otherwise, it will install MLflow from pip.
+    """
+    docker_cmds = ["RUN pip install azureml-sdk"]
+
+    if mlflow_path is not None:
+        mlflow_install_cmd = "RUN pip install -e {mlflow_path}".format(
+            mlflow_path=_get_container_path(mlflow_path))
+    elif not mlflow_version.endswith("dev"):
+        mlflow_install_cmd = "RUN pip install mlflow=={mlflow_version}".format(
+            mlflow_version=mlflow_version)
+    else:
+        raise MlflowException(
+                "You are running a 'dev' version of MLflow: `{mlflow_version}` that cannot be"
+                " installed from pip. In order to build a container image, either specify the"
+                " path to a local copy of the MLflow GitHub repository or install a release version"
```

Comment:
Maybe say "provide the `mlflow_home` parameter or install..." Not obvious how to do the former.